File: Leetcode/HumbleNumbers/187repeated-dna-sequences/187repeated-dna-sequences.py

from collections import defaultdict
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        arr=[]
        L=10
        if(len(s)<=L):
            return []
        s1=defaultdict(int)
        for i in range(0,len(s)-L+1):
            s1[s[i:i+L]]+=1
            if(s1[s[i:i+L]]==2):
                arr.append(s[i:i+L])
        return arr
                

def main():
    k=Solution()
    s = "AAAAACCCCCAAAAACCCCCCAAAAAGGGTTT"
    t=k.findRepeatedDnaSequences(s)
    print("t="+str(t))
if __name__=="__main__":
    main()
                
            
            
            
            

File: Leetcode/HumbleNumbers/187repeated-dna-sequences/C++/187repeated-dna-sequences.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
using namespace std;
class Solution {
public:
    std::vector<std::string> findRepeatedDnaSequences(std::string s) {
        std::unordered_map<std::string,int>words;
        std::vector<std::string> result;
        int left=0;
        int n=s.size();
        for(int i=0;i<=n-10;i++)
        {
            std::string m=s.substr(i,10);
            words[m]++;
            if(words[m]==2)
            {
                result.emplace_back(m);
            }
        }
        return result;
    }
};

int main(int argc,char const *argv[])
{
    
    std::string n="AAAAACCCCCAAAAACCCCCCAAAAAGGGTTT";
    Solution s;

     auto result=s.findRepeatedDnaSequences(n);
    
    for(auto it:result)
    {
        std::cerr<<"it="<<it<<std::endl;
    }

    return 0;
}